### PR TITLE
Issue 118 - Python social authentication with Github

### PR DESCRIPTION
**Pull request by anuschka**
Python local authentication is for now a link on the Login/Profile page. To get it working you have to do two things:
1. In Github go to Personal Settings/Applications -> Register new application
Application name = TrackCatz
Homepage URL = http://localhost:8000/
Application Description = TrackCatz application
Application callback URL = http://localhost:8000/
2. In your local TrackCat directory create settings_local.py and add two lines:
SOCIAL_AUTH_GITHUB_KEY = 'enter Client ID from your github registered app '
SOCIAL_AUTH_GITHUB_SECRET = 'enter Client Secret from your github registered app'

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,11 @@
 Django==1.7.1
-wsgiref==0.1.2
 Pillow==2.7.0
+PyJWT==0.4.1
+argparse==1.2.1
```
Comment (goranche) [inline requirements.txt]: why is this needed?

**Comment by goranche** [comment]
> Please remove style.css from this PR. It does not belong here

so remove it!!

and again with the screenshots in the PR... why? you started describing things just fine, and then ... what?
also, this does not belong in a PR... if you need to add a description on how to get things going (creating app keys, for example), add a document (or modify an existing one, maybe README.md) to the tree, as that is information, that will be needed in the future, not just for the duration of this PR

**Comment by goranche** [comment]
I'm kind of puzzled why there are commits in this PR that clearly relate to the footer...  
(I have a hypothesis, though)

**Comment by ialja** [comment]
@anuschka We cannot easily remove files from a Pull request. A Pull request can be either merged or closed as it is, it's up to the author to make sure only the necessary files are included ;)

File: TrackCat/settings.py [hunk 2]
```diff
@@ -100,3 +108,15 @@
...
+LOGIN_URL = '/login/'
+LOGIN_REDIRECT_URL = '/'
+LOGIN_ERROR_URL = '/login-error/'
+SOCIAL_AUTH_NEW_USER_REDIRECT_URL = '/'
+SOCIAL_AUTH_ENABLED_BACKENDS = ('github')
+SOCIAL_AUTH_DEFAULT_USERNAME = 'new_social_auth_user'
+SOCIAL_AUTH_GITHUB_KEY = ''
+SOCIAL_AUTH_GITHUB_SECRET = ''
+
+from settings_local import *
```
Comment (ialja) [inline TrackCat/settings.py]: You already have this import a [few lines above](https://github.com/anuschka/TrackCat/blob/issue-118-social-auth/TrackCat/settings.py#L107-L110) - it's best to keep that as the last part of the file and move the AUTH, LOGIN and SOCIAL stuff above with other authentication settings. See the [codeweekeu settings file](https://github.com/codeeu/coding-events/blob/master/codeweekeu/settings.py) for some organization tips and the way you can add comments, so we'll know what belongs where.

File: requirements.txt (hunk 1 above)
Comment (ercchy) [inline requirements.txt]: Actually these were dependencies that python-social-auth app needs and installed.

**Comment by anuschka** [comment]
I copied style.css from local branch master to local branch issue-118-social-auth and then pushed it to the upstream branch.

**Comment by goranche** [comment]
> you can just copy the whole thing

this in general is not the way to do it, because by doing that we might lose the history of this file, it could look as if @anuschka created it from scratch  (and if you take a look at [her repository](https://github.com/anuschka/TrackCat/blame/issue-118-social-auth/static/css/style.css) this is actually the case, information about the other contributing authors is lost)

the right way (if a revert of a previous commit isn't an option) is to **merge** the file back from the master branch, **not** copy it over after removing it

my recommendation is to reject this with extreme prejudice (I'm obviously not in a good mood :wink: )

**Comment by anuschka** [comment]
Sorry I do not understand what to do at this point. If there is a way how to fix the conflict please advice. IMHO the only other way is to delete my branch issue-118-social-auth and start from scratch...

**Comment by goranche** [comment]
don't do anything too hasty now... I'm not sure how, but I think there might be a way to fix it, but I'll probably try it out tomorrow

**Comment by anuschka** [comment]
Okeydokey :+1:

**Comment by ialja** [comment]
@goranche Most of the [history remains intact](https://github.com/anuschka/TrackCat/commits/issue-118-social-auth/static/css/style.css )- and we can actually merge this PR now (it's just be good to test it out before doing that).

**Comment by goranche** [comment]
> then pushed it to the upstream branch

no, you didn't, or at least I hope you didn't because that would mean you have mixed up upstream/origin in your local repository

@anuschka, can you explain what you tried to do with this commit: "Remove style.css from this PR"? ok, delete the file, but why?
you said you wanted to remove it from the PR, not the repository

**Comment by goranche** [comment]
> Most of the history remains intact

No, it doesn't, we lose information about contributions to the file, the other authors are removed from the history, check the blame screen I linked to in my comment

**Comment by ialja** [comment]
@anuschka Please [revert](https://www.atlassian.com/git/tutorials/undoing-changes/git-revert) to [this commit](https://github.com/anuschka/TrackCat/commit/fd453814643ed03f830db9359384141d598b0184), before the removal of the file took place.

**Comment by anuschka** [comment]
Please let me know if this is ok so I can make changes to readme.MD, settings.py and login.html that got wiped out with the revert.

**Comment by ialja** [comment]
@anuschka If you check the Files changes tab you'll see that the CSS still isn't right ... Might actually be a good idea to sync your master and start a fresh branch (keep the current one for reference). And please don't make changes to the CSS file unless they're related to this PR! :)

**Comment by anuschka** [comment]
OK :sweat:
